### PR TITLE
memberships api filter by tag

### PR DESCRIPTION
**Pull request by andrewsunglaekim**

### CODE REVIEW

**Comment by jshawl** [comment]
:+1:

File: app/controllers/api/memberships_controller.rb [hunk 1]
```diff
@@ -4,13 +4,16 @@ class MembershipsController < ApplicationController
     def index
       @cohort = Cohort.find(params[:cohort_id])
       if can? :list_cohort_members, @cohort
-        @names = @cohort.memberships.map do |member|
-          member.user.name
+        @members = @cohort.memberships
+        if params[:tag]
+          @members = @members.filter_by_tag(params[:tag])
         end
+        @names = @members.map{|member| member.user.name}
```
Comment (adambray) [inline app/controllers/api/memberships_controller.rb]: should we render memberships, and include names?

File: app/models/membership.rb [hunk 2]
```diff
@@ -56,4 +56,9 @@ def average_observations
     self.observations.any? ? average : nil
   end
 
+  def self.filter_by_tag tag
```
Comment (adambray) [inline app/models/membership.rb]: How is this working as a `self` (aka class) method?

Comment (adambray) [inline app/models/membership.rb]: ohhhh, it's on a result set.

**Comment by adambray** [comment]
:+1:

**Pull request merged**